engine/order_manager_test.go

```diff
@@ -645,6 +649,37 @@ func TestSubmit(t *testing.T) {
 	}
 }
 
+// TestSubmitOrderAlreadyInStore ensures that if an order is submitted, but the WS sees the conf before processSubmittedOrder
+// then we don't error that it was there already, and we merge together the details
+func TestSubmitOrderAlreadyInStore(t *testing.T) {
+	m := OrdersSetup(t)
+	submitReq := &order.Submit{
+		Type:      order.Market,
+		Pair:      btcusdPair,
+		AssetType: asset.Spot,
+		Side:      order.Buy,
+		Amount:    1,
+		Price:     1,
+		Exchange:  testExchange,
+	}
+	submitResp, err := submitReq.DeriveSubmitResponse("batman.obvs")
+	assert.Nil(t, err, "Deriving a SubmitResp should not error")
+
+	id, err := uuid.NewV4()
+	assert.Nil(t, err, "uuid should not error")
+	d, err := submitResp.DeriveDetail(id)
+	assert.Nil(t, err, "Derive Detail should not error")
+
+	d.ClientOrderID = "SecretSquirrelSauce"
+	assert.Nil(t, m.orderStore.add(d), "Adding an order should not error")
```

Comment:
```suggestion
        err = m.orderStore.add(d)
	assert.Nil(t, err, "Adding an order should not error")
```
I find this more readable

Reply:
Thanks. I flipflopped on this so happy to take a subjective view.